pew/pew.py

```diff
@@ -373,11 +373,23 @@ def setvirtualenvproject(env, project):
 
 
 def setproject_cmd():
-    """Given a virtualenv directory and a project directory, set the virtualenv up to be associated with the project."""
-    env = os.environ.get('VIRTUAL_ENV', args.get(1))
-    project = args.get(2, os.path.abspath('.'))
-    if not env:
+    """Given a virtualenv directory and a project directory, set the
+    virtualenv up to be associated with the project."""
+    if len(args) == 4:
+        env = args.get(2)
+        project = args.get(3)
+    elif len(args) == 3:
+        env = os.environ['VIRTUAL_ENV']
```

Comment:
This could give a KeyError if invoked with too few args outside of a virtualenv.
`os.environ.get('VIRTUAL_ENV')` would let the code get to the `if not env` check